ReactWindows/ReactNative.Shared/UIManager/NativeViewHierarchyManager.cs

```diff
@@ -571,32 +569,53 @@ public void DispatchCommand(int reactTag, int commandId, JArray args)
         /// view (shadow views cannot be anchors).
         /// </param>
         /// <param name="items">The menu items as an array of strings.</param>
+        /// <param name="error">Called if the popup menu failed to be shown.</param>
         /// <param name="success">
         /// A callback used with the position of the selected item as the first
         /// argument, or no arguments if the menu is dismissed.
         /// </param>
-        public void ShowPopupMenu(int tag, string[] items, ICallback success)
+        public void ShowPopupMenu(int tag, string[] items, ICallback error, ICallback success)
         {
 #if WINDOWS_UWP
             AssertOnCorrectDispatcher();
-            var view = ResolveView(tag);
-
-            var menu = new PopupMenu();
-            for (var i = 0; i < items.Length; ++i)
+            object view;
+            try
             {
-                menu.Commands.Add(new UICommand(
-                    items[i],
-                    cmd =>
-                    {
-                        success.Invoke(cmd.Id);
-                    },
-                    i));
+                view = ResolveView(tag);
+            }
+            catch (InvalidOperationException e)
+            {
+                error.Invoke(e.Message);
+                return;
             }
-#endif
 
-            // TODO: figure out where to popup the menu
-            // TODO: add continuation that calls the callback with empty args
+            var menu = new Windows.UI.Xaml.Controls.MenuFlyout();
```

Comment:
put using at the top in the appropriate #if section and eliminate fully qualified name.